### DISCUSSION TITLE
Historical Data access - Python OPCUA

### DISCUSSION THREAD

**Discussion by ksrangini05** [FreeOpcUa/python-opcua#1255] [Q&A] [answered]

**Comment by AndreasHeine** [comment, marked as answer]
moved to Discussions!

**Reply by AndreasHeine** [reply]
please remember we are not getting money for this and we are not a programing school! knowing the concepts of opcua is key to write good and reliable applications but thats the job of a software-developer...

**Reply by ksrangini05** [reply]
I wanted only suggestions. Apologize for the confusion.

**Reply by AndreasHeine** [reply]
you need to solve it on client side... the server can only provide the timeseries in a historizing variable!
if subscription/connection breaks you can try to recover with a history read from last known datachange notification timestamp

for this you should read more about OPC UA: Subscription and OPC UA: HistoricalAccess (not the old "OPC HDA" com/dcom stuff) then you can work out how to combine them!

https://github.com/FreeOpcUa/python-opcua/tree/master/examples

**Reply by ksrangini05** [reply]
thank you

**Reply by zerox1212** [reply]
> please remember we are not getting money for this and we are not a programing school! knowing the concepts of opcua is key to write good and reliable applications but thats the job of a software-developer...

Can we put this text in every issue and discussion template lol!
